app/components/learnergroup-list.js

```diff
@@ -1,84 +1,82 @@
 
-  sortedAscending: computed('sortBy', function() {
-    return this.sortBy.search(/desc/) === -1;
-  }),
-
-  sortedLearnerGroups: computed('learnerGroups.[]', 'sortBy', 'sortedAscending', function () {
-    if (!this.learnerGroups) {
-      return [];
-    }
-    let sortBy = this.sortBy;
-    if (sortBy.indexOf(':') !== -1) {
-      sortBy = sortBy.split(':', 1)[0];
-    }
+  get sortBy() {
+    return this.args.sortBy ?? this.localSortBy;
+  }
 
-    let sortedLearnerGroups = this.learnerGroups.sortBy(sortBy);
+  get sortedAscending(){
+    return this.sortBy.search(/desc/) === -1;
+  }
 
-    if (!this.sortedAscending) {
-      sortedLearnerGroups = sortedLearnerGroups.slice().reverse();
-    }
+  @action
+  cancelRemove(learnerGroup) {
+    this.toRemove = this.toRemove.filter(({ id }) => id !== learnerGroup.id);
+  }
 
-    return sortedLearnerGroups;
-  }),
+  @task
+  *confirmRemove(learnerGroup) {
+    this.preparingToRemove = [...this.preparingToRemove, learnerGroup];
+    const deletableGroup = yield this.createDeletableGroup(learnerGroup);
+    this.toRemove = [...this.toRemove, deletableGroup];
+    this.preparingToRemove = this.preparingToRemove.filter(lg => lg !== learnerGroup);
+  }
 
-  init() {
-    this._super(...arguments);
-    this.set('learnerGroupsForCopy', []);
-    this.set('learnerGroupsForRemovalConfirmation', []);
-  },
+  @action
+  cancelCopy(learnerGroup) {
+    this.toCopy = this.toCopy.filter(lg => lg !== learnerGroup);
+  }
 
-  actions: {
-    cancelRemove(learnerGroup) {
-      this.learnerGroupsForRemovalConfirmation.removeObject(learnerGroup);
-    },
+  @action
+  startCopy(learnerGroup) {
+    this.toCopy = [...this.toCopy, learnerGroup];
+  }
 
-    confirmRemove(learnerGroup) {
-      const canDelete = this.canDelete;
-      if (canDelete) {
-        this.learnerGroupsForRemovalConfirmation.pushObject(learnerGroup);
-      }
-    },
+  @task
+  *copy(withLearners, learnerGroup) {
+    yield this.args.copy(withLearners, learnerGroup);
+    this.cancelCopy(learnerGroup);
+  }
 
-    cancelCopy(learnerGroup) {
-      this.learnerGroupsForCopy.removeObject(learnerGroup);
-    },
+  @action
+  setSortBy(what) {
+    if(this.sortBy === what){
+      what += ':desc';
+    }
+    if (this.args.setSortBy) {
+      this.args.setSortBy(what);
+    }
+    this.localSortBy = what;
+  }
 
-    startCopy(learnerGroup) {
-      this.learnerGroupsForCopy.pushObject(learnerGroup);
-    },
+  async getCoursesForGroup(learnerGroup) {
+    const offerings = (await learnerGroup.offerings).toArray();
+    const ilms = (await learnerGroup.ilmSessions).toArray();
+    const arr = [].concat(offerings, ilms);
 
-    async copy(withLearners, learnerGroup) {
-      const copy = this.copy;
-      await copy(withLearners, learnerGroup);
-      this.learnerGroupsForCopy.removeObject(learnerGroup);
-    },
+    const sessions = await Promise.all(arr.mapBy('session'));
+    const filteredSessions = sessions.filter(Boolean).uniq();
```

Comment:
what is this ork mischief?!

Reply:
Not my fault, I learned it from @Trott 😆. Allegedly `Boolean` is a function that returns a bool. Fancy right?